lib/shelf_io.dart

```diff
@@ -74,8 +74,10 @@ Future handleRequest(HttpRequest request, Handler handler) async {
   try {
     shelfRequest = _fromHttpRequest(request);
   } catch (error, stackTrace) {
```

Comment:
Hmm, we probably have to distinguish between errors that happen because of a bug in `_fromHttpRequest` and errors that happen because the request is malformed.

Reply:
Ack, updated the PR: only return 400 when URL-validation check failed, otherwise keep the 500.

Reply:
SGTM